Security review PR #350: resolving comments

Zitzak opened the pull request:
## Implements
- closes #356 
- closes #357
- closes #358
- closes #359

Zitzak committed:
fix(gh-356): add view modifier to functions for gas optimization

The following function are set to view to optimize gas:
- _issueTokensFormulaWrapper
- _issueTokens
- _redeemTokensFormulaWrapper
- _redeemTokens

Zitzak committed:
docs(gh-357): update comments for clarity and misspelling

Zitzak committed:
perf(gh-359): Remove unnecessary payable modifier

Removed payable modifer because the function will not receive
Ether, and removing them makes for gas optimization

Zitzak committed:
perf(gh-358): remove unnecessary input validation for fee percentage

The function _calculateFeeDeductedDepositAmount can only be called from
a function which has already done an input validation. For that reason
the value can never be bigger than BPS. The check is unnecessary.

Removed the test which check for the input validation

Zitzak committed:
refactor(gh-360): Rename function sellOrder(for) & buyOrder(for)

Because there is not really an order placed but instead tokens are
being bought and sold on the bonding curve, the word "order" can be
misleading. The functions have been renamed for clarity

Zitzak committed:
Merge remote-tracking branch 'origin/dev' into security_review/PR#350

Zitzak committed:
refactor(gh-361): remove unnecessary function _issueTokens & _redeemTokens

Zitzak committed:
refactor(gh-374): move token decimal input validation to implementation

The issuance token decimal input validation was done within the
BondingCurveFundingManagerBase. It made more sense to move this to the
implementation contract. In addition to the refactor a check was added
within the BancorVirtualSupplyBondingCurveFundingManager for tokens
to revert tokens decimals < 7 to ensure no destructive precision loss
when using PPM for calculations.

0xNuggan commented on test/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.t.sol:
```diff
@@ -1060,6 +1060,31 @@ contract BancorVirtualSupplyBondingCurveFundingManagerTest is ModuleTest {
+    }
+
+    function testSetDecimals(uint8 _newDecimals) public {
+        vm.assume(_newDecimals > 7);
```
needs to be `>=` for completeness' sake

0xNuggan commented on test/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.t.sol:
```diff
@@ -1060,6 +1060,31 @@ contract BancorVirtualSupplyBondingCurveFundingManagerTest is ModuleTest {
+    /* Test _setDecimals function
+        ├── When decimal is set to lower than seven
+        |   └── it should revert
+        └── when decimal is bigger than zero
```
It should be `equal or bigger than seven`

0xNuggan commented on src/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.sol:
```diff
@@ -341,6 +337,25 @@ contract BancorVirtualSupplyBondingCurveFundingManager is
 
+    /// @dev Sets the number of decimals for the token.
+    /// This function overrides the internal function set in BondingCurveFundingManagerBase, adding
+    /// an input validation specific for the Bancor Formula untilizing implementation, after which
```
typo

Zitzak committed:
fix: address pr review comments

- Fix typo in function comment
- Fix branching test tree comment
- Fix test to include vm.assume for == 7
